app/controllers/concerns/audit_log.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+module AuditLog
+  extend ActiveSupport::Concern
+
+  private
+
+  class Audit
+    VALID_METHODS = [:info, :warn, :debug, :error]
+
+    def self.log(user, action, *args)
+      logger(:info, user, action, *args)
+    end
+
+    def self.info(user, action, *args)
+      logger(:info, user, action, *args)
+    end
+
+    def self.warn(user, action, *args)
+      logger(:warn, user, action, *args)
+    end
+
+    def self.debug(user, action, *args)
+      logger(:debug, user, action, *args)
+    end
+
+    def self.error(user, action, *args)
+      logger(:error, user, action, *args)
+    end
+
+    def self.logger(level, user, action, *args)
+      if AUDIT_LOGGER && VALID_METHODS.include?(level)
```

Comment:
atm a invalid call would silently be ignored ... not good ...
`raise ArgumentError unless VALID_METHODS.include?(level)`